File: src/main/java/com/company/service/SubjectService.java
package com.company.service;

import com.company.entities.Subject;
import com.company.repository.SubjectRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class SubjectService {

    @Autowired
    private SubjectRepository repository;

    public List<Subject> findAll(){
        List<Subject> subjects = (ArrayList<Subject>)repository.findAll();
        return subjects.stream().sorted(Comparator.comparing(Subject::getId)).collect(Collectors.toList());
    }

    public void addSubject(Subject subject){
        repository.save(subject);
    }

    public Subject findSubjectById(long id){
        Subject subject = repository.findById(id).get();
        return subject;
    }

    public void removeSubject(long id){
        repository.delete(findSubjectById(id));
    }

    public void updateSubject(Subject subject){
        repository.save(subject);
    }

}

File: src/main/resources/static/app/service/subject.service.js
angular
    .module("SchoolApp")
    .factory("SubjectService", ["$http", function ($http) {
        var ALL_SUBJECTS = "http://localhost:8080/subject/findAll";
        var ADD_SUBJECT = "http://localhost:8080/subject/add";
        var UPDATE_SUBJECT = "http://localhost:8080/subject/update";
        var DELETE_SUBJECT = "http://localhost:8080/subject/remove/";

        var service = {};


        service.getAllSubjects = function getAllSubjects() {
            return $http.get(ALL_SUBJECTS);
        };

        service.addSubject = function addSubject(subject) {
            return $http.post(ADD_SUBJECT, subject)
        };

        service.updateSubject = function updateSubject(subject) {
            return $http.post(UPDATE_SUBJECT, subject)
        };

        service.deleteSubject = function deleteSubject(subject) {
            return $http.delete(DELETE_SUBJECT + subject.id)
        };

        return service;
    }]);